### ISSUE TITLE
Making a Slack workspace available only for one VS Code workspace

### ISSUE THREAD

**Issue by kylrth** [karigari/vscode-chat#402]
There is a Slack workspace I use only when I work with a particular subset of my VS Code workspaces. I'd like to not be connected to Slack when I open a different workspace, one unrelated to that Slack workspace. Is this possible currently?

**Comment by arjun27** [comment]
Thanks @kylrth. This is not possible currently; the Slack authentication stuff is independent of the loaded workspace. When you say "particular subset", what does that mean? Are these some particular repos?

**Comment by kylrth** [comment]
Yeah I use VS Code to work on some personal repos, and I don't want to see my work Slack when I'm working on them.

**Comment by arjun27** [comment]
Thanks. We could add a workspace setting to enable Slack. I don't think I will be able to prioritize this for now, but happy to accept PRs for this

**Comment by kylrth** [comment]
I would love it if this could allow different workspaces to have different Slack workspaces. Say I had a Slack workspace for my job, and one for some open source work I'm involved in. If I open a workspace in VS Code, the associated Slack workspace should open.

**Comment by arjun27** [comment]
Yeah, that is how I was thinking. Currently, the extension maintains state about chat (Slack) workspaces in `localStorage`. This can be matched against repo-level settings, which could define something like

```
chat.enabledWorkspaces: {
   type: "slack",
   id: "workspace-id"
}
```

Also related: #11
